src/main/java/com/artipie/VertxMain.java

```diff
@@ -152,6 +167,65 @@ public static void main(final String... args) throws IOException, ParseException
         new VertxMain(config, vertx, port).start();
     }
 
+    /**
+     * Find repos and start them on specific port.
+     * @param settings The settings
+     * @throws IOException In case of error reading settings.
+     * @throws InterruptedException If interrupted.
+     */
+    private void findAndStartOnSpecificPorts(final Settings settings)
+        throws IOException, InterruptedException {
+        final ArtipieRepositories repositories = new ArtipieRepositories(settings);
+        final BlockingStorage blocking = new BlockingStorage(settings.storage());
+        new ArrayList<>(blocking.list(Key.ROOT))
+            .stream()
+            .filter(
+                key -> {
+                    final String full = key.string();
+                    final int index = full.lastIndexOf('/');
+                    final String fname = index == -1 ? full : full.substring(index);
+                    return !fname.startsWith("_")
+                        && (fname.endsWith(".yml") || fname.endsWith(".yaml"));
+                }
+            )
+            .map(
+                key -> {
+                    final int sport;
+                    try {
+                        sport = Yaml.createYamlInput(new String(blocking.value(key)))
+                            .readYamlMapping()
+                            .yamlMapping("repo")
+                            .integer("port");
+                        return new Pair<>(sport, key);
+                    } catch (final InterruptedException | IOException exc) {
+                        throw new IllegalStateException(exc);
+                    }
+                }
+            ).filter(pair -> pair.getValue0() > 0)
+            .forEach(
+                pair -> {
+                    try {
+                        final Integer sport = pair.getValue0();
+                        final Key key = new Key.From(
+                            pair.getValue1().string()
+                                .replaceAll("\\.yaml", "")
+                                .replaceAll("\\.yml", "")
+                        );
+                        final Slice slice = repositories.slice(key);
+                        new VertxSliceServer(this.vertx, slice, sport).start();
+                        Logger.info(
+                            VertxMain.class,
+                            "Slice at %s started at port:%d",
+                            key.string(),
+                            sport
+                        );
+                    } catch (final IOException exc) {
+                        throw new IllegalStateException(exc);
+                    }
+                }
+            );
+    }
```

Comment:
@Sammers21 this method is to complex to understand and contains a lot of logic taken from other classes. Please extract this logic to separate class so it could be understood